### PR TITLE
add stages support for build

### PR DESCRIPTION
**Pull request by tanhe123**

### CODE REVIEW

File: lib/taskflows/copy-source-task.js [hunk 1]
```diff
@@ -12,8 +13,12 @@ class CopySourceTask extends Task {
     this.excludeFiles = excludeFiles;
   }
 
-  async run() {
-    await copySourceTo(this.sourceDir, this.targetDir, this.excludeFiles);
+  async run() {    
+    if (this.sourceDir === this.targetDir) {
+      log.debug('skipping copies, this.sourceDir %s, this.targetDir %s', this.sourceDir, this.targetDir);
```
Comment (vangie) [inline lib/taskflows/copy-source-task.js]: Skipping copy， the source dir is %s, the target dir is %s

File: test/taskflows/python/pip/pip-install-task.test.js [hunk 2]
```diff
@@ -30,6 +31,7 @@ describe('test PipInstallTask', () => {
     await pipInstallTask.run();
 
     assert.calledWith(fs.pathExists, path.join(sourceDir, 'requirements.txt'));
-    assert.calledWith(cmd.execCommand, 'pip', ['install', '--disable-pip-version-check', '-t', artifactDir, '-r', path.join(sourceDir, '/requirements.txt')]);
+    assert.calledWith(cmd.execCommand, 'pip', ['install', '--user', '-r', path.join(sourceDir, '/requirements.txt')]);
+    assert.calledWith(fs.ensureDir, '/artifactDir/.fun/python');
```
Comment (vangie) [inline test/taskflows/python/pip/pip-install-task.test.js]: 目录名不要搞驼峰吧 artifactDir => artifacts

**Review by vangie — CHANGES_REQUESTED** [review]

File: lib/taskflows/copy-source-task.js (hunk 1 above)
Comment (tanhe123) [reply lib/taskflows/copy-source-task.js]: done

File: test/taskflows/python/pip/pip-install-task.test.js (hunk 2 above)
Comment (tanhe123) [reply test/taskflows/python/pip/pip-install-task.test.js]: done

**Review by vangie — APPROVED** [review]

**Pull request merged**